tests/test_manage.py

```diff
@@ -286,6 +345,133 @@ def test_create_request_with_granting_access(client):
+    [
+        {
+            # provide resource_path to get the default reader policies
+            "username": "requestor_user",
+            "resource_path": "/study/123456",
+            "resource_id": "uniqid",
+            "resource_display_name": "My Resource",
+        },
+        {
+            # provide resource_paths without role_ids to get the default reader policies
+            "username": "requestor_user",
+            "resource_paths": ["/study/123456", "/mds_gateway", "/cedar"],
+            "resource_id": "uniqid",
+            "resource_display_name": "My Resource",
+        },
+        {
+            # resource_paths will take precedence over resource_path
+            "username": "requestor_user",
+            "resource_paths": ["/study/123456", "/mds_gateway", "/cedar"],
+            "resource_path": "/older_study/000111",
+            "resource_id": "uniqid",
+            "resource_display_name": "My Resource",
+        },
+    ],
+)
+def test_create_request_with_resource_path(client, data):
+    fake_jwt = "1.2.3"
+
+    if data.get("resource_paths"):
+        policy_id = get_auto_policy_id_for_resource_paths(data["resource_paths"])
+    elif data.get("resource_path"):
+        # resource_path will get cast to a list
+        policy_id = get_auto_policy_id_for_resource_paths([data["resource_path"]])
+    else:
+        policy_id = None
+
+    res = client.post(
+        "/request", json=data, headers={"Authorization": f"bearer {fake_jwt}"}
+    )
+    assert res.status_code == 201, res.text
+    request_data = res.json()
+    request_id = request_data.get("request_id")
+    assert request_id, "POST /request did not return a request_id"
+    assert request_data == {
+        "request_id": request_id,
+        "username": data["username"],
+        "policy_id": policy_id,
+        "resource_id": data["resource_id"],
+        "resource_display_name": data["resource_display_name"],
+        "status": config["DEFAULT_INITIAL_STATUS"],
+        # just ensure revoke, created_time and updated_time are there:
+        "revoke": False,
+        "created_time": request_data["created_time"],
+        "updated_time": request_data["updated_time"],
+    }
+
+    # get the request
+    res = client.get(f"/request/{request_id}")
+    assert res.status_code == 200, res.text
+    assert res.json() == request_data
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        {
+            # include single role_ids and single resource_paths
+            "username": "requestor_user",
+            "role_ids": ["study_registrant"],
+            "resource_paths": ["/study/123456"],
+            "resource_id": "uniqid",
+            "resource_display_name": "My Resource",
+        },
+        {
+            # include multiple role_ids and resource_paths
+            "username": "requestor_user",
+            "role_ids": ["study_registrant", "/mds_user", "/cedar_user"],
+            "resource_paths": ["/study/123456", "/mds_gateway", "/cedar"],
```

Comment:
Just thinking, even though it is not absolutely necessary, was wondering if we can add another nested `resource_path`, just so that it would be easier for any future reference. 

```python
"resource_paths": ["/study/123456", "/study/7890", "/mds_gateway", "/cedar"],
```

Reply:
Yes, this is confusing. 

- the comments have been updated with additional resource_paths
- unit tests have been added to show more examples

In both places, the references to `cedar` have been made more generic.